src/gluonts/mx/trainer/callback.py

```diff
@@ -265,121 +264,50 @@ def include(
-            callback.on_validation_batch_end(training_network=training_network)
+    def on_validation_batch_end(self, *args: Any, **kwargs: Any) -> None:
+        self._exec_all("on_validation_batch_end", *args, **kwargs)
 
-    def on_train_epoch_end(
-        self,
-        epoch_no: int,
-        epoch_loss: float,
-        training_network: nn.HybridBlock,
-        trainer: gluon.Trainer,
-    ) -> bool:
-        return np.all(
-            [
-                callback.on_train_epoch_end(
-                    epoch_no=epoch_no,
-                    epoch_loss=epoch_loss,
-                    training_network=training_network,
-                    trainer=trainer,
-                )
-                for callback in self.callbacks
-            ]
-        )
+    def on_train_epoch_end(self, *args: Any, **kwargs: Any) -> bool:
+        return all(self._exec_all("on_train_epoch_end", *args, **kwargs))
 
-    def on_validation_epoch_end(
-        self,
-        epoch_no: int,
-        epoch_loss: float,
-        training_network: nn.HybridBlock,
-        trainer: gluon.Trainer,
-    ) -> bool:
-        return np.all(
-            [
-                callback.on_validation_epoch_end(
-                    epoch_no=epoch_no,
-                    epoch_loss=epoch_loss,
-                    training_network=training_network,
-                    trainer=trainer,
-                )
-                for callback in self.callbacks
-            ]
-        )
+    def on_validation_epoch_end(self, *args: Any, **kwargs: Any) -> bool:
+        return all(self._exec_all("on_validation_epoch_end", *args, **kwargs))
 
-    def on_epoch_end(
-        self,
-        epoch_no: int,
-        epoch_loss: float,
-        training_network: nn.HybridBlock,
-        trainer: gluon.Trainer,
-        best_epoch_info: Dict[str, Any],
-        ctx: mx.Context,
-    ) -> bool:
-        return np.all(
-            [
-                callback.on_epoch_end(
-                    epoch_no=epoch_no,
-                    epoch_loss=epoch_loss,
-                    training_network=training_network,
-                    trainer=trainer,
-                    best_epoch_info=best_epoch_info,
-                    ctx=ctx,
-                )
-                for callback in self.callbacks
-            ]
-        )
+    def on_epoch_end(self, *args: Any, **kwargs: Any) -> bool:
+        return all(self._exec_all("on_epoch_end", *args, **kwargs))
 
-    def on_train_end(
-        self,
-        training_network: nn.HybridBlock,
-        temporary_dir: str,
-        ctx: mx.context.Context = None,
-    ) -> None:
-        for callback in self.callbacks:
-            callback.on_train_end(
-                training_network=training_network,
-                temporary_dir=temporary_dir,
-                ctx=ctx,
-            )
+    def on_train_end(self, *args: Any, **kwargs: Any) -> None:
+        self._exec_all("on_train_end", *args, **kwargs)
+
+    def _exec_all(
```

Comment:
I think just `_exec` sounds better to me. `all` could imply that you actually also run `all` on the result.

Reply:
Also, I think it's better to have `_exec_all` over the `on_xxx` methods.

Reply:
What do you mean with the second comment?

Reply:
I.e. put `_exec_all` after `def include`.

That way we have a separation between the special methods of the class and the list of `on_<something` methods.